server/svix-server/src/v1/endpoints/application.rs

```diff
@@ -176,7 +177,7 @@ async fn update_application(
 async fn delete_application(
     Extension(ref db): Extension<DatabaseConnection>,
     Path(app_id): Path<ApplicationIdOrUid>,
-    permissions: Permissions,
+    AuthenticatedOrganization { permissions }: AuthenticatedOrganization,
```

Comment:
Why not also change this to AuthenticatedOrganizationWithApp and remove the manual fetch inside like you did above?